node/src/command/solochain.rs

```diff
@@ -0,0 +1,419 @@
+                .with_id("test_id")
+                .with_chain_type(ChainType::Development)
+                .with_genesis_config_patch(Default::default())
+                .build(),
+        ),
+        wasm_method: Default::default(),
+        wasmtime_precompiled: None,
+        wasm_runtime_overrides: None,
+        rpc_addr: None,
+        rpc_max_connections: 0,
+        rpc_cors: None,
+        rpc_methods: Default::default(),
+        rpc_max_request_size: 0,
+        rpc_max_response_size: 0,
+        rpc_id_provider: None,
+        rpc_max_subs_per_conn: 0,
+        rpc_port: 0,
+        rpc_message_buffer_capacity: 0,
+        rpc_batch_config: BatchRequestConfig::Disabled,
+        rpc_rate_limit: None,
+        rpc_rate_limit_whitelisted_ips: vec![],
+        rpc_rate_limit_trust_proxy_headers: false,
+        prometheus_config: None,
+        telemetry_endpoints: None,
+        default_heap_pages: None,
+        offchain_worker: Default::default(),
+        force_authoring: false,
+        disable_grandpa: false,
+        dev_key_seed: None,
+        tracing_targets: None,
+        tracing_receiver: Default::default(),
+        max_runtime_instances: 0,
+        announce_block: false,
+        data_path: Default::default(),
+        base_path,
+        informant_output_format: Default::default(),
+        runtime_cache_size: 0,
+    }
+}
+
+/// Returns the default path for configuration directory based on the chain_spec
+pub(crate) fn build_solochain_config_dir(base_path: &PathBuf) -> PathBuf {
+    // base_path:  Collator1000-01/data/containers
+    // config_dir: Collator1000-01/data/config
+    let mut base_path = base_path.clone();
+    base_path.pop();
+
+    base_path.join("config")
+}
+
+pub fn keystore_config(
+    keystore_params: Option<&sc_cli::KeystoreParams>,
+    config_dir: &PathBuf,
+) -> sc_cli::Result<KeystoreConfig> {
+    keystore_params
+        .map(|x| x.keystore_config(config_dir))
+        .unwrap_or_else(|| Ok(KeystoreConfig::InMemory))
+}
+
+/// Get the zombienet keystore path from the solochain collator keystore.
+fn zombienet_keystore_path(keystore: &KeystoreConfig) -> PathBuf {
+    let keystore_path = keystore.path().unwrap();
+    let mut zombienet_path = keystore_path.to_owned();
+    // Collator1000-01/data/config/keystore/
+    zombienet_path.pop();
+    // Collator1000-01/data/config/
+    zombienet_path.pop();
+    // Collator1000-01/data/
+    zombienet_path.push("chains/simple_container_2000/keystore/");
+    // Collator1000-01/data/chains/simple_container_2000/keystore/
+
+    zombienet_path
+}
+
+/// When running under zombienet, collator keys are injected in a different folder from what we
+/// expect. This function will check if the zombienet folder exists, and if so, copy all the keys
+/// from there into the expected folder.
+pub fn copy_zombienet_keystore(keystore: &KeystoreConfig) -> std::io::Result<()> {
+    let keystore_path = keystore.path().unwrap();
```

Comment:
What is the reasoning behind changing the key path?

Reply:
In zombienet, the keystore path depends on the chain id, which I already decided to ignore. So I'm hardcoding `simple_container_2000` as the chain id in this case. I tried changing the keystore path in zombienet config but that is not possible.

Outside of zombienet, the default keystore path is `$base_path/chains/dancebox_local/keystore/`, which has the `dancebox_local` chain id but that's confusing because this is not dancebox. So I have set the keystore path to `$base_path/config/keystore/`.

Essentially we need the keystore to be in a separate folder outside of `containers`, because that gets deleted when the collator rotates. The exact path does not matter, and can be changed by the user by using `--keystore-path`.

Reply:
Make sense.

*disregard*